### PR TITLE
ENH: Implement io.nullable_backend config for read_parquet

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] xref #48957 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v1.6.0.rst [hunk 1]
```diff
@@ -33,6 +33,7 @@ Other enhancements
 - :meth:`Series.add_suffix`, :meth:`DataFrame.add_suffix`, :meth:`Series.add_prefix` and :meth:`DataFrame.add_prefix` support an ``axis`` argument. If ``axis`` is set, the default behaviour of which axis to consider can be overwritten (:issue:`47819`)
 - :func:`assert_frame_equal` now shows the first element where the DataFrames differ, analogously to ``pytest``'s output (:issue:`47910`)
 - Added new argument ``use_nullable_dtypes`` to :func:`read_csv` to enable automatic conversion to nullable dtypes (:issue:`36712`)
+- Added new global configuration, ``io.nullable_backend`` to allow ``use_nullable_dtypes=True`` to return pyarrow-backed dtypes when set to ``"pyarrow"`` in :func:`read_parquet` (:issue:`48957`)
```
Comment (phofl) [inline doc/source/whatsnew/v1.6.0.rst:36]: Should engine default to arrow if the option Is set to arrow?

Comment (mroeschke) [reply doc/source/whatsnew/v1.6.0.rst:36]: I'd be a little hesitant to override a user specifying `pd.read_*(..., engine="not-arrow")` and this option forcing arrow to be used instead

Comment (phofl) [reply doc/source/whatsnew/v1.6.0.rst:36]: Agreed, but we could make engine=NoDefault and only override if not given. 


But I agree that we should consider this carefully

File: pandas/tests/io/test_parquet.py [hunk 2]
```diff
@@ -1021,6 +1021,43 @@ def test_read_parquet_manager(self, pa, using_array_manager):
         else:
             assert isinstance(result._mgr, pd.core.internals.BlockManager)
 
+    def test_read_use_nullable_types_pyarrow_config(self, pa, df_full):
```
Comment (phofl) [inline pandas/tests/io/test_parquet.py:1024]: This is failing in the min versions build

Comment (mroeschke) [reply pandas/tests/io/test_parquet.py:1027]: Thanks, fixed

**Comment by mroeschke** [comment]
@phofl any follow-ups here?

File: pandas/tests/io/test_parquet.py [hunk 3]
```diff
@@ -1021,6 +1021,46 @@ def test_read_parquet_manager(self, pa, using_array_manager):
         else:
             assert isinstance(result._mgr, pd.core.internals.BlockManager)
 
+    @pytest.mark.xfail(
+        pa_version_under2p0, reason="Timezone conversion incorrect for pyarrow < 2.0"
```
Comment (phofl) [inline pandas/tests/io/test_parquet.py:1025]: Can remove this

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @mroeschke

**Pull request merged**